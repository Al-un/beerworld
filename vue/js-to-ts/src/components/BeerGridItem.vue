<template>
  <div class="beer-grid-item">
    <img class="beer-img" :src="beer.mainPicture" />
    <div class="beer-name">{{ beer.name }}</div>
  </div>
</template>

<script>
export default {
  props: {
    beer: { type: Object, required: true },
  },
};
</script>

<style lang="scss">
.beer-grid-item {
  .beer-img {
    width: 100%;
  }
}
</style>
