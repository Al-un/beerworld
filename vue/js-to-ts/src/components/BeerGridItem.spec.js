import { mount } from "@vue/test-utils";

import BeerGridItem from "./BeerGridItem";

let wrapper;

describe("components/BeerGridItem", () => {
  it("renders and match snapshot", () => {
    wrapper = mount(BeerGridItem, {
      propsData: {
        beer: { name: "beer1", mainPicture: "mainPic1" },
      },
    });

    expect(wrapper).toMatchSnapshot();
  });
});
