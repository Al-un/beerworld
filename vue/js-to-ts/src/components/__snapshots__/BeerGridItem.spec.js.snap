// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/BeerGridItem renders and match snapshot 1`] = `
<div class="beer-grid-item"><img src="mainPic1" class="beer-img">
  <div class="beer-name">beer1</div>
</div>
`;
